### ISSUE TITLE
3.14 build error after recent commit: `pylifecycle.c:50:12: fatal error: os/log.h: No such file or directory`

### ISSUE THREAD

**Issue by barracuda156** [python/cpython#128146]
# Bug report

### Bug description:

The recent commit https://github.com/python/cpython/commit/2041a95e68ebf6d13f867e214ada28affa830669 has broken the build of Python for a number of macOS versions, since `<os/log.h>` is not guaranteed to exist, but has been included unconditionally.

```
Python/pylifecycle.c:50:12: fatal error: os/log.h: No such file or directory
   50 | #  include <os/log.h>
      |            ^~~~~~~~~~
compilation terminated.
make: *** [Python/pylifecycle.o] Error 1
make: *** Waiting for unfinished jobs....
```

@freakboy3742 Could you please take a look at this?

### CPython versions tested on:

3.14

### Operating systems tested on:

macOS

### Linked PRs
* gh-128165
* gh-128575

**Comment by freakboy3742** [comment]
@barracuda156 When you say "A number of macOS versions" - can you clarify which ones you mean? I'm guessing the availability issue is the one referred to by [this comment](https://github.com/python/cpython/pull/127592/files#diff-69223f5bb01f359a23246afbbf75b45c622e719424418d658674fbb6c2b36fadR2960); in which case, the fix is likely using the same #ifdef gate on the `os/log.h` import.

**Comment by barracuda156** [comment]
@freakboy3742 Yes, I believe, that is correct: https://groups.google.com/a/chromium.org/g/crashpad-dev/c/5CY_M9R6mIw

**Comment by freakboy3742** [comment]
@barracuda156 Presumptive fix in #128165; it's currently failing CI for reasons that I can't explain, and I can't reproduce locally.

**Comment by barracuda156** [comment]
@freakboy3742 Thank you, I will run the build and let you know.

**Comment by hugovk** [comment]
PR merged and backported, thanks all!
